dotenv/mod.ts

```diff
@@ -0,0 +1,234 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+import { difference, removeEmptyValues } from "./util.ts";
+
+export interface DotenvConfig {
+  [key: string]: string;
+}
+
+export interface ConfigOptions {
+  path?: string;
+  export?: boolean;
+  safe?: boolean;
+  example?: string;
+  allowEmptyValues?: boolean;
+  defaults?: string;
+}
+
+interface ParseResult {
+  env: DotenvConfig;
+  exports: Set<string>;
+}
+
+const emptyParseResult = (): ParseResult => ({ env: {}, exports: new Set() });
+const EXPORT_REGEX = /^\s*export\s+/;
+
+export function parse(rawDotenv: string): ParseResult {
+  const env: DotenvConfig = {};
+  const exports = new Set<string>();
+
+  for (const line of rawDotenv.split("\n")) {
+    if (!isAssignmentLine(line)) continue;
+    const lhs = line.slice(0, line.indexOf("=")).trim();
+    const { key, exported } = parseKey(lhs);
+    if (exported) {
+      exports.add(key);
+    }
+    let value = line.slice(line.indexOf("=") + 1).trim();
+    if (hasSingleQuotes(value)) {
+      value = value.slice(1, -1);
+    } else if (hasDoubleQuotes(value)) {
+      value = value.slice(1, -1);
+      value = expandNewlines(value);
+    } else value = value.trim();
+    env[key] = value;
+  }
+
+  return { env, exports };
+}
+
+const defaultConfigOptions = {
+  path: `.env`,
+  export: false,
+  safe: false,
+  example: `.env.example`,
+  allowEmptyValues: false,
+  defaults: `.env.defaults`,
+};
+
+export function config(options: ConfigOptions = {}): DotenvConfig {
```

Comment:
I would suggest to flip the naming - `configSync`/`config` instead of `config` and `configAsync` - this would match most of built-in `Deno` APIs